src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -680,6 +680,12 @@ public ErrorDiagnostic ArgumentCountMismatch(int argumentCount, int mininumArgum
                 "An empty indexer is not allowed. Specify a valid expression."
             );
 
+            public Diagnostic RuntimePropertyNotAllowed(string property) => new ErrorDiagnostic(

```

Comment:
Right now, indexing via string interpolation also triggers this exception. Open to raising some other exception if it'd be perceived as misleading